packages/core-data/src/reducer.js

```diff
@@ -183,6 +183,26 @@ export function themeGlobalStyleVariations( state = {}, action ) {
 	return state;
 }
 
+const withMultiEntityRecordEdits = ( reducer ) => ( state, action ) => {
+	if ( action.type === 'UNDO' || action.type === 'REDO' ) {
+		const { stackedEdits } = action;
+
+		let newState = state;
```

Comment:
Should we shallow copy the state here?

Reply:
No, it's fine, this is a HoC, the lower level reducers will copy the state if needed.